homeassistant/components/zwave_js/entity.py

```diff
@@ -61,19 +61,31 @@ def device_info(self) -> dict:
             "identifiers": {get_device_id(self.client, self.info.node)},
         }
 
+    def generate_name(
+        self, value_name: str, additional_info: Optional[List[str]] = None
+    ) -> str:
+        """Generate entity name."""
+        if additional_info is None:
+            additional_info = []
+        node_name = self.info.node.name or self.info.node.device_config.description
+        name = f"{node_name}: {value_name}"
+        for item in additional_info:
+            if item:
+                name += f" - {item}"
+        # append endpoint if > 1
+        if self.info.primary_value.endpoint > 1:
+            name += f" ({self.info.primary_value.endpoint})"
+
+        return name
+
     @property
     def name(self) -> str:
         """Return default name from device name and value name combination."""
-        node_name = self.info.node.name or self.info.node.device_config.description
-        value_name = (
+        return self.generate_name(
             self.info.primary_value.metadata.label
```

Comment:
I didn't check but isn't this part the same everywhere ?

Reply:
no, in other places we use property_name or command_class_value as the part of the name immediately after node name

Reply:
what we can do is make this optional and default to the current approach